### ISSUE TITLE
Enable OL tests for partial implementation of Object Lock

### ISSUE THREAD

**Issue by amwolff** [storj/edge#406]
## Goal

This issue is to make meaningful progress on Object Lock implementation within core and edge services.

## What needs to be done? • Acceptance Criteria

We only need to implement a small subset of Object Lock (for example, Legal Hold is out of scope at this moment) first. Let's find tests in test suites that we use that can be enabled for what we already implemented.

## Depends on…

At least #397.

## Links

- an easy-to-read and insightful introduction to Object Lock: [Using S3 Object Lock](https://docs.aws.amazon.com/AmazonS3/latest/userguide/object-lock.html)

**Comment by neo-cypher** [comment]
Ceph object lock tests enabled in gateway-st , not main, currently failing. Further review is needed.

**Comment by neo-cypher** [comment]
Report of failing tests and review of tests enabled underway.

**Comment by pwilloughby** [comment]
We're only able to pass one addtional ceph test, `test_object_lock_put_obj_retention_invalid_mode`.  The remaining tests are relying on setup code using  put_object_lock_configuration.  I think we should wait on this ticket until https://github.com/storj/edge/issues/501 is complete.

**Comment by ferristocrat** [comment]
> We're only able to pass one addtional ceph test, `test_object_lock_put_obj_retention_invalid_mode`. The remaining tests are relying on setup code using put_object_lock_configuration. I think we should wait on this ticket until #501 is complete.

bummer. but makes sense - we have initial validation from manual tests and associated integration tests...

**Comment by ferristocrat** [comment]
We initially used this ticket to test our early version of object lock that we used to validate Veeam compatibility. We are updating the ticket to include enabling ceph tests for our now complete version of object lock.

**Comment by halkyon** [comment]
https://review.dev.storj.io/c/storj/gateway/+/15402 enables the rest of the object lock tests. I raised a few new issues for the backlog to fix the remaining problems it found:

* https://github.com/storj/edge/issues/528
* https://github.com/storj/edge/issues/529
